pkg/api/controller.go

```diff
@@ -1103,7 +1104,22 @@ func (c *Controller) CreateRepository(w http.ResponseWriter, r *http.Request, bo
 		return
 	}
 
-	err := ensureStorageNamespaceRW(ctx, c.BlockAdapter, body.StorageNamespace)
+	parsedNs, err := url.ParseRequestURI(body.StorageNamespace)
```

Comment:
Not sure why this is not part of the ensure* call. I assume we'd want the same error detection if we added another "create repo" operation. 
If the issue is better error reporting from that call then we might want to do it there. E.g. Return an internal error code and an external message.

Reply:
Agree with Ariel, this logic belongs to the ensureStorageNamespaceRW method because it's part of ensuring the validity of the storage namespace. Can you please move it there?

Reply:
nit; it's up to you but I prefer the name parsedNamespace for readability reasons